applications/StructuralMechanicsApplication/custom_constitutive/hyper_elastic_isotropic_neo_hookean_plane_strain_2d.cpp

```diff
@@ -0,0 +1,150 @@
+
+void HyperElasticIsotropicNeoHookeanPlaneStrain2D::GetLawFeatures(Features& rFeatures)
+{
+    //Set the type of law
+    rFeatures.mOptions.Set( PLANE_STRAIN_LAW );
+    rFeatures.mOptions.Set( FINITE_STRAINS );
+    rFeatures.mOptions.Set( ISOTROPIC );
+
+    //Set strain measure required by the consitutive law
+    rFeatures.mStrainMeasures.push_back(StrainMeasure_GreenLagrange);
+    rFeatures.mStrainMeasures.push_back(StrainMeasure_Deformation_Gradient);
+    
+    //Set the strain size
+    rFeatures.mStrainSize = GetStrainSize();
+
+    //Set the spacedimension
+    rFeatures.mSpaceDimension = WorkingSpaceDimension();
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void HyperElasticIsotropicNeoHookeanPlaneStrain2D::CalculateConstitutiveMatrix(
+    Matrix& ConstitutiveMatrix,
+    const Matrix& InverseCTensor,
+    const double& DeterminantF,
+    const double& LameLambda,
+    const double& LameMu
+    )
+{
+    ConstitutiveMatrix.clear();
+    
+    const double log_j = std::log(DeterminantF);
+    
+    for(unsigned int i = 0; i < 3; i++)
+    {
+        for(unsigned int j = 0; j < 3; j++)
+        {
+            const unsigned int& i0 = this->msIndexVoigt2D3C[i][0];
+            const unsigned int& i1 = this->msIndexVoigt2D3C[i][1];
+            const unsigned int& j0 = this->msIndexVoigt2D3C[j][0];
+            const unsigned int& j1 = this->msIndexVoigt2D3C[j][1];
+            
+            ConstitutiveMatrix(i,j) = (LameLambda*InverseCTensor(i0,i1)*InverseCTensor(j0,j1)) + ((2.0 * LameMu-LameLambda * 2.0 * log_j) * 0.5 * (InverseCTensor(i0,j0) * InverseCTensor(i1,j1) + InverseCTensor(i0,j1) * InverseCTensor(i1,j0)));
+        }
+    }
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void HyperElasticIsotropicNeoHookeanPlaneStrain2D::CalculateCauchyGreenStrain(
+    Parameters& rValues,
+    Vector& rStrainVector
+    )
+{
+    //1.-Compute total deformation gradient
+    const Matrix& F = rValues.GetDeformationGradientF();
+
+    // e = 0.5*(inv(C) - I)
+    Matrix C_tensor = prod(trans(F),F);
+    
+    rStrainVector[0] = 0.5 * ( C_tensor( 0, 0 ) - 1.00 );
+    rStrainVector[1] = 0.5 * ( C_tensor( 1, 1 ) - 1.00 );
+    rStrainVector[2] = C_tensor( 0, 1 );
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void HyperElasticIsotropicNeoHookeanPlaneStrain2D::CalculateAlmansiStrain(
+    Parameters& rValues,
+    Vector& rStrainVector
+    )
+{
+    //1.-Compute total deformation gradient
+    const Matrix& F = rValues.GetDeformationGradientF();
+
+    // e = 0.5*(1-inv(B))
+    Matrix B_tensor = prod(F,trans(F));
```

Comment:
this is normally called C

Reply:
C is the right Cauchy Green tensor, this is the left Cauchy Green tensor, also called B

Reply:
https://en.wikipedia.org/wiki/Finite_strain_theory?wprov=sfla1